features/FxA-45-device-registration-api/FX-DESKTOP.md

```diff
@@ -0,0 +1,122 @@
+Firefox Device Registration
+===========================
+
+This is a breakdown of the states Firefox needs to handle/worry about.
+
+See the [README.md](README.md) file for a list of requirements and user
+stories.
+
+## New profile, no FxA deviceID, no FxA sessionToken
+
+Users of a new Firefox profile who have never signed up/in to an FxA backed
+service will have neither an FxA sessionToken nor an FxA device ID.
+
+Firefox should take no action.
+
+## Existing profile, no FxA deviceID, has FxA sessionToken
+
+This is a user who has previously signed in to FxA to connect
+to Sync. This user will have an FxA sessionToken, but may not yet
+have an FxA deviceID.
+
+The user facing goal is for the device to appear in the user's device
+list without additional user action. To facilitate this, Firefox
+should register the device on behalf of the user.
+
+On startup, Firefox should attempt to register the device with the
+FxA auth server using the name and type currently stored in the
+user's Sync preferences.
+
+Firefox will send the following information to the FxA auth server to
+register the device:
+
+* deviceName (from Sync preferences)
+* deviceType (from Sync preferences)
+* pushCallback (from Push preferences)
+* pushPublicKey (from Push preferences)
+
+### Registration success
+
+The FxA auth server will respond with:
+
+* deviceID
+* deviceName
+* deviceType
+* pushCallback
+* pushPublicKey
+
+Firefox must then store the returned deviceID.
+
+### Registration failure
+
+If registration fails because of a network error, Firefox should re-attempt
```

Comment:
>  like an auth error

From a user-visible-errors point of view, it may be more like a "can't connect to the tokenserver" error, i.e. it prevents sync from syncing, but there's nothing the user can do to intervene.